src/vacuum-card.js

```diff
@@ -184,20 +186,52 @@ class VacuumCard extends LitElement {
       battery_level,
       battery_icon,
       friendly_name,
+      detected_pad,
+      tank_level,
     };
   }
 
+  renderMopSetting(source) {
+    const [mopBehavior, amount] = source.split('-').map((x) => x.trim());
```

Comment:
Also my vacuum is mopping and vacuum at the same. It has an extra attribute called `water_level` to specify the amount.
The format what you are using here is very specific for your use case (`{cleaning style}-{jet spray amount}`) and it must be also set as attribute fan speed. This would not work for my one.